### DISCUSSION TITLE
Exogenous variables transformation on a real-life set where test exogenous variables aren't available

### DISCUSSION THREAD

**Discussion by matsuobasho** [Nixtla/mlforecast#361] [General]
There is an assumption in the way that exogenous variables are currently set up that the exogenous variable data for predicted timestamp x will be available.  However, in my (and I suspect in many other real-life uses-cases), I don't have those future exogenous variables.  In my case, I want to create lags on exogenous variables, in which case even though I don't have future values, I will have the lagged values from the end of historic data.  I realize that in such a scenario, the use from including exogenous variables will be limited to the number of timestamps I'm lagging by, but I still want to try this.

Of course, I can manually create the lags like I'm intending, but aside from `transform_exog` is there any kind of function in `mlforecast` that achieves what I'm intending.  In essence, this function would take as inputs historic data, the forecast horizon and the lags and then create lagged variables from that (there would be a lot of nulls in the output).

**Comment by jmoralez** [comment]
Is [this](https://github.com/Nixtla/mlforecast/issues/328#issuecomment-2030009549) what you mean?

**Reply by matsuobasho** [reply]
Yes, exactly.  That code makes sense and I can see how it would be hard difficult to have a custom function for this.
